src/unittests/floating_point.spec.ts

```diff
@@ -89,81 +89,58 @@ interface ConstructorCase {
-    [
-      // Common cases
-      { input: [0, 10], expected: [0, 10] },
-      { input: [-5, 0], expected: [-5, 0] },
-      { input: [-5, 10], expected: [-5, 10] },
-      { input: [0], expected: [0] },
-      { input: [10], expected: [10] },
-      { input: [-5], expected: [-5] },
-      { input: [2.5], expected: [2.5] },
-      { input: [-1.375], expected: [-1.375] },
-      { input: [-1.375, 2.5], expected: [-1.375, 2.5] },
-
-      // Edges
-      { input: [0, kValue.f64.positive.max], expected: [0, kValue.f64.positive.max] },
-      { input: [kValue.f64.negative.min, 0], expected: [kValue.f64.negative.min, 0] },
-      { input: [kValue.f64.negative.min, kValue.f64.positive.max], expected: [kValue.f64.negative.min, kValue.f64.positive.max] },
-
-      // Infinities
-      { input: [0, kValue.f64.infinity.positive], expected: [0, Number.POSITIVE_INFINITY] },
-      { input: [kValue.f64.infinity.negative, 0], expected: [Number.NEGATIVE_INFINITY, 0] },
-      { input: [kValue.f64.infinity.negative, kValue.f64.infinity.positive], expected: kAnyBounds },
-
-      // Note: Out of range values are limited to infinities for abstract float,
-      // due to abstract float and 'number' both being f64.
-      // So there is no  separate OOR tests, the testing framework considers
-      // them duplicates.
-    ]
+g.test('constructor_f32af')
+  .params(u =>
+    u
+      .combine('trait', ['f32', 'abstract'] as const)
+      .beginSubcases()
+      .expandWithParams<ConstructorCase>(p => {
+        const constants = FP[p.trait].constants();
+        // prettier-ignore
+        const cases: ConstructorCase[] = [
+          // Common cases
+          { input: [0, 10], expected: [0, 10] },
+          { input: [-5, 0], expected: [-5, 0] },
+          { input: [-5, 10], expected: [-5, 10] },
+          { input: [0], expected: [0] },
+          { input: [10], expected: [10] },
+          { input: [-5], expected: [-5] },
+          { input: [2.5], expected: [2.5] },
+          { input: [-1.375], expected: [-1.375] },
+          { input: [-1.375, 2.5], expected: [-1.375, 2.5] },
+
+          // Edges
+          { input: [0, constants.positive.max], expected: [0, constants.positive.max] },
+          { input: [constants.negative.min, 0], expected: [constants.negative.min, 0] },
+          { input: [constants.negative.min, constants.positive.max], expected: [constants.negative.min, constants.positive.max] },
+
+          // Infinities
+          { input: [0, constants.positive.infinity], expected: [0, Number.POSITIVE_INFINITY] },
+          { input: [constants.negative.infinity, 0], expected: [Number.NEGATIVE_INFINITY, 0] },
+          { input: [constants.negative.infinity, constants.positive.infinity], expected: kAnyBounds },
+        ];
+
+        // Note: Out of range values are limited to infinities for abstract float, due to abstract
+        // float and 'number' both being f64. So there is no separate OOR tests for abstract float,
```

Comment:
So there is no separate OOR tests for abstract float, otherwise the testing framework will consider them as duplicates.

Reply:
Done